daemon/cmd/daemon.go

```diff
@@ -419,6 +422,52 @@ func NewDaemon(ctx context.Context, dp datapath.Datapath) (*Daemon, *endpointRes
 		bootstrapStats.k8sInit.End(true)
 	}
 
+	// The kube-proxy replacement should be initialized after establishing
+	// connection to kube-apiserver, but before starting a k8s watcher, as
+	// retrieving Node object for self is needed by BPF NodePort device selection,
+	// and the k8s watcher depends on option.Config.EnableNodePort flag which
+	// can be modified by the initialization routine.
+	initKubeProxyReplacementOptions()
+	if option.Config.EnableNodePort {
+		if err := node.InitNodePortAddrs(option.Config.Devices); err != nil {
+			log.WithError(err).Fatal("Failed to initialize NodePort addrs")
+		}
+	}
+	// BPF masquerade depends on BPF NodePort, so the following checks should
+	// happen after invoking initKubeProxyReplacementOptions().
+	if option.Config.Masquerade && option.Config.EnableBPFMasquerade {
+		// TODO(brb) nodeport + ipvlan constraints will be lifted once the SNAT BPF code has been refactored
+		if !option.Config.EnableNodePort {
+			log.Fatalf("BPF masquerade requires NodePort (--%s=\"true\")", option.EnableNodePort)
+		}
+		if option.Config.DatapathMode == datapathOption.DatapathModeIpvlan {
+			log.Fatalf("BPF masquerade works only in veth mode (--%s=\"%s\"", option.DatapathMode, datapathOption.DatapathModeVeth)
+		}
+		if option.Config.EgressMasqueradeInterfaces != "" {
+			log.Fatalf("BPF masquerade does not allow to specify devices via --%s. Use --%s instead.", option.EgressMasqueradeInterfaces, option.Devices)
```

Comment:
Is this basically because `--egress-masquerade-interfaces` receives an iptables wildcard parameter which can specify either directly a device or a range of devices with syntax like `eth+` and we don't want to support that? Or more just to avoid conflicting configuration from similar options?

Reply:
Oh, I guess this is already in the tree. Well I'm still curious.

Reply:
> receives an iptables wildcard parameter which can specify either directly a device or a range of devices with syntax like eth+ and we don't want to support that?

Yes, that's the reason. Once we decouple BPF NodePort from BPF masq (in v1.9), I'm planning to add a support for `--egress-masquerade-interfaces` with BPF masq.